src/frontend/komponenter/Fagsak/Behandle/FastsettVedtakProvider.tsx

```diff
@@ -0,0 +1,119 @@
+import * as React from 'react';
+import { IBarnBeregning, ordinærBeløp } from '../../../typer/behandle';
+import { IFagsak, sakstyper } from '../../../typer/fagsak';
+import { IFelt } from '../../../typer/felt';
+import { lagInitiellFelt } from '../../../typer/provider';
+import { erGyldigDato } from '../../../utils/validators';
+
+export enum actions {
+    SETT_BARNAS_BEREGNING = 'SETT_BARNAS_BEREGNING',
+    SETT_SAKSTYPE = 'SETT_SAKSTYPE',
+    SETT_BEHANDLINGSRESULTAT = 'SETT_BEHANDLINGSRESULTAT',
+    SETT_SENDER_INN = 'SETT_SENDER_INN',
+}
+
+export interface IAction {
+    payload?: any;
+    type: actions;
+}
+
+type Dispatch = (action: IAction) => void;
+
+export interface IState {
+    barnasBeregning: Array<IFelt<IBarnBeregning>>;
+    behandlingsresultat: string;
+    sakstype: string;
+    senderInn: boolean;
+}
+
+export const lastInitialState = (fagsak: IFagsak): IState => ({
+    barnasBeregning: fagsak.behandlinger[0].barnasFødselsnummer.map(barn => {
+        return lagInitiellFelt<IBarnBeregning>(
+            {
+                barn,
+                beløp: ordinærBeløp,
+                startDato: '',
+            },
+            erGyldigDato
+        );
+    }),
+    behandlingsresultat: 'innvilget',
+    sakstype: sakstyper.ORDINÆR.id,
+    senderInn: false,
+});
+
+const FastsettVedtakStateContext = React.createContext<IState | undefined>(undefined);
```

Comment:
Don't quite understand why we have to use context here, because they are only be used by FastsettVedtakSkjema component. Maybe made them state of the component?

Reply:
We also use this in FastsettVedtakSkjema. One could argue that we should send the state and dispatch to the children, but this would not invite coders to keep props to a minimum and refactor out components.